README.md

```diff
@@ -226,96 +226,68 @@ See the documentation for customizations and `ToolBar` examples.
+class _FlutterWindowDelegate extends NSWindowDelegate {
+  @override
+  void windowWillEnterFullScreen() {
+    WindowManipulator.removeToolbar();
+    super.windowWillEnterFullScreen();
   }
 
-  override func viewDidLoad() {
-    super.viewDidLoad()
-
-    self.addChild(flutterViewController)
-
-    flutterViewController.view.frame = self.view.bounds
-    flutterViewController.backgroundColor = .clear // **Required post-Flutter 3.7.0**
-    flutterViewController.view.autoresizingMask = [.width, .height]
-    self.view.addSubview(flutterViewController.view)
+  @override
+  void windowDidExitFullScreen() {
+    WindowManipulator.addToolbar();
+    super.windowDidExitFullScreen();
   }
 }
 
-class MainFlutterWindow: NSWindow, NSWindowDelegate {
-  override func awakeFromNib() {
-    delegate = self
-    let blurryContainerViewController = BlurryContainerViewController()
-    let windowFrame = self.frame
-    self.contentViewController = blurryContainerViewController
-    self.setFrame(windowFrame, display: true)
-
-    if #available(macOS 10.13, *) {
-      let customToolbar = NSToolbar()
-      customToolbar.showsBaselineSeparator = false
-      self.toolbar = customToolbar
-    }
-    self.titleVisibility = .hidden
-    self.titlebarAppearsTransparent = true
-    if #available(macOS 11.0, *) {
-      // Use .expanded if the app will have a title bar, else use .unified
-      self.toolbarStyle = .unified
-    }
-
-    self.isMovableByWindowBackground = true
-    self.styleMask.insert(NSWindow.StyleMask.fullSizeContentView)
-
-    self.isOpaque = false
-    self.backgroundColor = .clear
-
-    RegisterGeneratedPlugins(registry: blurryContainerViewController.flutterViewController)
-
-    super.awakeFromNib()
-  }
+/// This method initializes macos_window_utils and styles the window.
+Future<void> _initMacosWindowUtils() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await WindowManipulator.initialize(enableWindowDelegate: true);
+  await WindowManipulator.setMaterial(
+      NSVisualEffectViewMaterial.windowBackground);
+  await WindowManipulator.enableFullSizeContentView();
+  await WindowManipulator.makeTitlebarTransparent();
+  await WindowManipulator.hideTitle();
+  await WindowManipulator.addToolbar();
+
+  // Use NSWindowToolbarStyle.expanded if the app will have a title bar,
+  // otherwise use NSWindowToolbarStyle.unified.
+  await WindowManipulator.setToolbarStyle(
+      toolbarStyle: NSWindowToolbarStyle.unified);
+
+  // Create a delegate that removes the toolbar in full-screen mode.
+  final delegate = _FlutterWindowDelegate();
+  WindowManipulator.addNSWindowDelegate(delegate);
+
+  // Auto-hide toolbar and menubar in full-screen mode.
+  final options = NSAppPresentationOptions.from({
+    NSAppPresentationOption.fullScreen,
+    NSAppPresentationOption.autoHideToolbar,
+    NSAppPresentationOption.autoHideMenuBar,
+    NSAppPresentationOption.autoHideDock,
+  });
+  options.applyAsFullScreenPresentationOptions();
+}
```

Comment:
Again, I think stuff like this should be provided by the library.

Reply:
I chose to do it this way so that it was more faithful to how it was before the migration to macos_window_utils. Plus, I believe doing it this way may provide more customization options for the developer. If this functionality should be provided by the library, then I’d need to think about how to do that in a way that doesn’t sacrifice said customizability.

Reply:
Yes, I think we ought to provide classes that represent all possible use cases a developer could want.